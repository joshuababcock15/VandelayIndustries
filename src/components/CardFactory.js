import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { CardBody, CardTitle, CardSubtitle, Card, CardText } from 'reactstrap';

const propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  buildingName: PropTypes.string,
  city: PropTypes.string,
  country: PropTypes.string,
  state: PropTypes.string,
  streetLine: PropTypes.string,
  zipcode: PropTypes.string,
  linkRef: PropTypes.string,
  link: PropTypes.string,
};

const defaultProps = {};

const CardWrapper = styled.div`
  width: 100%;
  padding: 40px;
`;

const Link = styled.a`
  color: black;
`;

const ContentWrapper = styled.div`
  max-width: 500px;
`;

const CardFactory = (props) => {
  const {
    name,
    description,
    buildingName,
    city,
    country,
    state,
    streetLine,
    zipcode,
    linkRef,
    link,
  } = props;

  return (
    <CardWrapper>
      <Card color="light">
        <ContentWrapper>
          <CardBody>
            <CardTitle tag="h5">{name}</CardTitle>
            <CardText>{description}</CardText>
            <CardSubtitle className="mb-2 text-muted" tag="h6">
              {buildingName}
            </CardSubtitle>
            <CardSubtitle className="mb-2 text-muted" tag="h6">
              {streetLine} {zipcode}
            </CardSubtitle>
            <CardSubtitle className="mb-2 text-muted" tag="h6">
              {city} {country} {state}
            </CardSubtitle>
            <Link href={linkRef}>{link}</Link>
          </CardBody>
        </ContentWrapper>
      </Card>
    </CardWrapper>
  );
};

CardFactory.propTypes = propTypes;
CardFactory.defaultProps = defaultProps;

export default CardFactory;
