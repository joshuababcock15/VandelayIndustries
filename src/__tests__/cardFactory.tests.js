import React from 'react';
import renderer from 'react-test-renderer';
import CardFactory from '../components/CardFactory';

describe('CardFactory', () => {
  const tree = renderer.create(<CardFactory name="Newark Latex Mfg." />);
  const testInstance = tree.toJSON();

  test('Matches snapshot', () => {
    expect(testInstance).toMatchSnapshot();
  });

  test('default props', () => {
    const testRenderer = renderer.create(
      <CardFactory
        name="Newark Latex Mfg."
        description="Key East Coast facility"
        buildingName="Newark Latex"
        city="Bronx"
        country="USA"
        state="NY"
        streetLine="173 E 161 St"
        zipcode="10451"
        link="Learn more"
        linkRef="/home"
      />
    );
    const testInstances = testRenderer.root;
    expect(testInstances.props.name).toBe('Newark Latex Mfg.');
    expect(testInstances.props.description).toBe('Key East Coast facility');
    expect(testInstances.props.buildingName).toBe('Newark Latex');
    expect(testInstances.props.city).toBe('Bronx');
    expect(testInstances.props.country).toBe('USA');
    expect(testInstances.props.state).toBe('NY');
    expect(testInstances.props.streetLine).toBe('173 E 161 St');
    expect(testInstances.props.zipcode).toBe('10451');
    expect(testInstances.props.link).toBe('Learn more');
    expect(testInstances.props.linkRef).toBe('/home');
  });
});
