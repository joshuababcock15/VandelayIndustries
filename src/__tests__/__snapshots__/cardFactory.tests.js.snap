// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFactory Matches snapshot 1`] = `
<div
  className="sc-bdvvtL ctqOxA"
>
  <div
    className="card bg-light"
  >
    <div
      className="sc-dkPtRN jeZswX"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Newark Latex Mfg.
        </h5>
        <p
          className="card-text"
        />
        <h6
          className="mb-2 text-muted card-subtitle"
        />
        <h6
          className="mb-2 text-muted card-subtitle"
        >
           
        </h6>
        <h6
          className="mb-2 text-muted card-subtitle"
        >
          , 
           
        </h6>
        <a
          className="sc-gsDKAQ eSZZXj"
        />
      </div>
    </div>
  </div>
</div>
`;
